### ISSUE TITLE
xmpp user with multiple devices keeps repeating "changed their display name"

### ISSUE THREAD

**Issue by emorrp1** [matrix-org/matrix-bifrost#280]
**Describe the bug**
I didn't report this at the time, as we can't identify the root cause, but it's now been independently confirmed in a completely different room. You can see from [this public conversation](https://view.matrix.org/room/!qLhNfILESSCaasbRWB:freedombox.emorrp1.name/?anchor=$whmSpvjQNm8FGfQYyFNxoCJ_XxsoN2h5Ky6mlCRC3yk&offset=-29) (and older) that one user keeps rotating between poVoq/kris, when they've tried to set their display name in the ordinary way on the XMPP side.

My guess is that this is actually to do with Presence, and there's something going on with XMPP's equivalent of /away messages, but that's completely unfounded, sorry for the lack of Reproduction steps. Besides, these messages are simply annoying to see constantly in Element clients.

> matrix.org offers an xmpp bridge that you can connect to as any other xmpp MUC. I just got reports that my nick kept cycling between detrout/diane over and over and I'm not really sure why

**Expected behavior**
An XMPP user's preferred display name, when different from their JID should be stable until they manually change it. One workaround that appears to work is for them to set the display name to be equal to their JID user part.

**Comment by detrout** [comment]
I also had the problem and I wasn't sure if it was JID != nick or if I had multiple clients connected to the bridge and I had different nicks set in the different clients.

**Comment by dora71** [comment]
Although this bug is already older, the problem still persists. Can confirm it with blabber.im Messenger (XMPP) and various Matrix rooms.
Will test if the workaround is ok here.

**Comment by ple1n** [comment]
![image](https://user-images.githubusercontent.com/68096727/174529939-0dc605a1-85b8-47ad-8ac2-4600d56def1e.png)

@_xmpp_blumedesdevils=[40yax.im](http://40yax.im/):[matrix.org](http://matrix.org/)

I have seen this issue in different rooms. It persists.

**Comment by Half-Shot** [comment]
Yeah, this looks like it's basically picking the device name and bouncing between them. We should probably just pick *a* device name and stick to it.

**Comment by endorxmr** [comment]
I don't know if this is the same issue or not, but there are two xmpp users in a bridged room that keep spamming a ton of name changes/leave/rejoin events. Even with notifications disabled, it's getting extremely annoying and it's spamming the room a lot.

<img width="344" alt="xmpp_rejoining_1" src="https://user-images.githubusercontent.com/28117225/224627570-2319c60e-5d1f-4c6f-bde3-98500b34d8da.PNG">

<img width="680" alt="xmpp_rejoining_2" src="https://user-images.githubusercontent.com/28117225/224628015-394c3748-e0bf-4d7a-b4b2-0cf42ada6865.PNG">

**Comment by MichaelSasser** [comment]
We would kindly appreciate, if this issue could be resolved or at least mitigated. We have a large room with 12.5k users and get asked a lot about XMPP user's name bouncing as well as the joining/leaving and if we could "just ban them" as this would be "annoying" and "frustrating". We are trying to be inclusive and welcoming to all of our users in the same way, but this issue becomes unbearable. Our "top XMPP user" currently generates 500 membership events per week (see graph below). Therefore, we are leaning towards temporarily banning XMPP users who generate a high number of state events until this issue is resolved, even though we don't want to. 

Please do something.<br /><br />

![XMPP membership events](https://github.com/matrix-org/matrix-bifrost/assets/4131336/8a48ded8-560e-485f-93da-a2b628d9616f)

---

Room: `!iuyQXswfjgxQMZGrfQ:matrix.org`

**Comment by Neustradamus** [comment]
Have you progressed on this ticket?
